sway-lsp/src/core/token.rs

```diff
@@ -24,15 +26,20 @@ use tower_lsp::lsp_types::{Position, Range};
 pub enum AstToken {
     Declaration(Declaration),
     Expression(Expression),
+    StructExpression(StructExpression),
     StructExpressionField(StructExpressionField),
     StructScrutineeField(StructScrutineeField),
     FunctionDeclaration(FunctionDeclaration),
     FunctionParameter(FunctionParameter),
+    FunctionApplicationExpression(FunctionApplicationExpression),
+    MethodApplicationExpression(MethodApplicationExpression),
+    AmbiguousPathExpression(AmbiguousPathExpression),
+    DelineatedPathExpression(DelineatedPathExpression),
+    AbiCastExpression(AbiCastExpression),
     StructField(StructField),
     EnumVariant(EnumVariant),
     TraitFn(TraitFn),
     TraitConstraint(TraitConstraint),
-    ConstantDeclaration(ConstantDeclaration),
```

Comment:
Just checking, is there a reason `ConstantDeclaration` was removed here?

Reply:
Good question, we are still collecting these but wrapping them under the `Declaration` variant above. This way we can easy look up all the declaration tokens and then match on them from there.